src/clients/AcrossConfigStoreClient.ts

```diff
@@ -0,0 +1,440 @@
+import {
+  spreadEvent,
+  sortEventsDescending,
+  spreadEventWithBlockNumber,
+  paginatedEventQuery,
+  EventSearchConfig,
+  utf8ToHex,
+  MakeOptional,
+  toBN,
+  max,
+  sortEventsAscending,
+} from "../utils";
+
+import { Contract, BigNumber } from "ethers";
+import winston from "winston";
+
+import {
+  L1TokenTransferThreshold,
+  TokenConfig,
+  GlobalConfigUpdate,
+  ParsedTokenConfig,
+  SpokeTargetBalanceUpdate,
+  SpokePoolTargetBalance,
+  RouteRateModelUpdate,
+  ConfigStoreVersionUpdate,
+  DisabledChainsUpdate,
+} from "../interfaces";
+
+import * as lpFeeCalculator from "../lpFeeCalculator";
+import { BlockFinder, across } from "@uma/sdk";
+import { HubPoolClient } from "./AcrossHubPoolClient";
+
+export const GLOBAL_CONFIG_STORE_KEYS = {
+  MAX_RELAYER_REPAYMENT_LEAF_SIZE: "MAX_RELAYER_REPAYMENT_LEAF_SIZE",
+  MAX_POOL_REBALANCE_LEAF_SIZE: "MAX_POOL_REBALANCE_LEAF_SIZE",
+  VERSION: "VERSION",
+  DISABLED_CHAINS: "DISABLED_CHAINS",
+};
+
+export class AcrossConfigStoreClient {
+  public readonly blockFinder;
+
+  public cumulativeRateModelUpdates: across.rateModel.RateModelEvent[] = [];
+  public cumulativeRouteRateModelUpdates: RouteRateModelUpdate[] = [];
+  public cumulativeTokenTransferUpdates: L1TokenTransferThreshold[] = [];
+  public cumulativeMaxRefundCountUpdates: GlobalConfigUpdate[] = [];
+  public cumulativeMaxL1TokenCountUpdates: GlobalConfigUpdate[] = [];
+  public cumulativeSpokeTargetBalanceUpdates: SpokeTargetBalanceUpdate[] = [];
+  public cumulativeConfigStoreVersionUpdates: ConfigStoreVersionUpdate[] = [];
+  public cumulativeDisabledChainUpdates: DisabledChainsUpdate[] = [];
+
+  private rateModelDictionary: across.rateModel.RateModelDictionary;
+  public firstBlockToSearch: number;
+
+  public hasLatestConfigStoreVersion = false;
+
+  public isUpdated = false;
+
+  constructor(
+    readonly logger: winston.Logger,
+    readonly configStore: Contract,
+    readonly hubPoolClient: HubPoolClient,
+    readonly eventSearchConfig: MakeOptional<EventSearchConfig, "toBlock"> = { fromBlock: 0, maxBlockLookBack: 0 },
+    protected readonly configOverride: {
+      enabledChainIds: number[];
+      defaultConfigStoreVersion: number;
+    } = { enabledChainIds: [], defaultConfigStoreVersion: 0 }
+  ) {
```

Comment:
Added an override here so that we don't need to reference constant variables from the relayer. Throughout the rest of the code, I've replaced calls like `DEFAULT_CONFIG_STORE_VERSION` to `this.configOverride.defaultConfigStoreVersion`;

Reply:
+1